packages/squeak/src/pages/api/roadmap/unsubscribe.ts

```diff
@@ -0,0 +1,39 @@
+import type { NextApiRequest, NextApiResponse } from 'next'
+
+import { notAuthenticated, safeJson } from 'src/lib/api/apiUtils'
+import nextConnect from 'next-connect'
+import { corsMiddleware, allowedOrigin } from 'src/lib/middleware'
+import { getSessionUser } from 'src/lib/auth'
+import prisma from 'src/lib/db'
+import { addPersonToCustomerIO } from 'src/util/customerIO'
+
+const handler = nextConnect<NextApiRequest, NextApiResponse>()
+    .use(corsMiddleware)
+    .use(allowedOrigin)
+    .post(unsubscribeFromRoadmapItem)
+
+async function unsubscribeFromRoadmapItem(req: NextApiRequest, res: NextApiResponse) {
+    const user = await getSessionUser(req)
+    if (!user) return notAuthenticated(res)
+
+    const { id } = req.body
+
+    if (!id) return res.status(500).json({ error: 'No roadmap ID provided' })
```

Comment:
```suggestion
    if (!id) return res.status(400).json({ error: 'No roadmap ID provided' })
```

`400` is probably better here since the request body is invalid